http/RequestParser.cpp

```diff
@@ -197,6 +223,85 @@ void RequestParser::parseBodyContentLength(u_int8_t const& octet) {
   }
 }
 
+// chunk body 파싱
+void RequestParser::parseBodyChunked(u_int8_t const& octet) {
+  switch (_status) {
+    case BODY_CHUNKED:
+      _status = BODY_CHUNK_SIZE;
+      // fallthrough
+    case BODY_CHUNK_SIZE:
+      parseBodyChunkSize(octet);
+      break;
+    case BODY_CHUNK_DATA:
+      parseBodyChunkData(octet);
+      break;
+    case BODY_CHUNK_TRAILER:
+      parseBodyChunkTrailer(octet);
+      break;
+    default:
+      throw std::runtime_error(
+          "[2302] RequestParser: parseBodyChunked - invalid status");
+  }
+}
+
+// chunk body에 대한 chunk-size 파싱
+// - CRLF가 입력되었을 경우 입력 종료 후 chunk-size 저장
+// - chunk-size가 0인 경우 last-chunk로 처리 후
+//   입력받은 body를 request 객체에 저장
+// - chunk-size가 1 이상인 경우 chunk-data를 읽도록 상태 변경
+void RequestParser::parseBodyChunkSize(u_int8_t const& octet) {
+  _chunkSizeBuffer.push_back(octet);
+
+  if (octet == '\n' and isEndWithCRLF(_chunkSizeBuffer)) {
+    processBodyChunkSize();
+    _status = _chunkSize == 0 ? BODY_CHUNK_TRAILER : BODY_CHUNK_DATA;
+    _chunkSizeBuffer.clear();
+
+    if (_status == BODY_CHUNK_TRAILER) {
+      std::string processResult = processBody();
+      _request.storeBody(processResult);
+    }
+  }
+}
+
+// chunk body에 대한 chunk-data 파싱
+// - chunk-size 만큼 chunk-data를 읽은 후 CRLF가 입력되지 않았을 경우 예외 발생
+void RequestParser::parseBodyChunkData(u_int8_t const& octet) {
+  const size_t crlfLength = 2;
+
+  const size_t bodySize = _body.size();
```

Comment:
저희 코드 컨벤션에 맞게 자료형 뒤에 const가 붙어야 할 것 같아요!

```suggestion
  size_t const crlfLength = 2;

  size_t const bodySize = _body.size();
```